### DISCUSSION TITLE
How to handle structures with partial occupancies ?

### DISCUSSION THREAD

**Discussion by BaptisteHeesooPy** [materialsproject/pymatgen#2880] [Q&A]
Good morning,

To study high-entropy perovskites such as Ba0.5Ca0.5FeO3, I am currently leveraging two amazing tools that you created, namely the M3Gnet and Pymatgen. However, I have discovered that cif files describing structures with partial occupancies, as it can be the case for high-entropy perovskites, cannot be analysed directly. Indeed, the function get_atoms in Pymatgen (https://github.com/materialsproject/pymatgen/blob/v2023.2.28/pymatgen/io/ase.py#L41-L131 (line 42)) raises an error if the structure is not ordered. Would you know a way to circumvent this issue ?

I thank you very much in advance for your time and help.

Best regards,

Baptiste

**Comment by shyuep** [comment]
pls see https://matgenb.materialsvirtuallab.org/ for notebook on ordering disordered structures. Most tools don't work with partially occupied structures. While M3GNet can work with partial occupancies, there may be some adjustment to the input code needed.

**Reply by BaptisteHeesooPy** [reply]
I thank you very much for your quick and helpful response Professor.
